Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -1497,7 +1488,378 @@ The hooked function outputted the decrypted string. You extracted the secret str
+0x000000009b2dc000 - 0x000000009b361000 rw- /dev/ashmem/dalvik-/data/app/sg.vantagepoint.helloworldjni-1/oat/arm64/base.art (deleted)
+0x000000009b361000 - 0x000000009b36e000 --- /dev/ashmem/dalvik-/data/app/sg.vantagepoint.helloworldjni-1/oat/arm64/base.art (deleted)
+0x000000009b36e000 - 0x000000009b371000 rw- /dev/ashmem/dalvik-/data/app/sg.vantagepoint.helloworldjni-1/oat/arm64/base.art (deleted)
+0x0000007d103be000 - 0x0000007d10686000 r-- /data/app/sg.vantagepoint.helloworldjni-1/oat/arm64/base.vdex
+0x0000007d10dd0000 - 0x0000007d10dee000 r-- /data/app/sg.vantagepoint.helloworldjni-1/oat/arm64/base.odex
+0x0000007d10dee000 - 0x0000007d10e2b000 r-x /data/app/sg.vantagepoint.helloworldjni-1/oat/arm64/base.odex
+0x0000007d10e3a000 - 0x0000007d10e3b000 r-- /data/app/sg.vantagepoint.helloworldjni-1/oat/arm64/base.odex
+0x0000007d10e3b000 - 0x0000007d10e3c000 rw- /data/app/sg.vantagepoint.helloworldjni-1/oat/arm64/base.odex
+0x0000007d1c499000 - 0x0000007d1c49a000 r-x /data/app/sg.vantagepoint.helloworldjni-1/lib/arm64/libnative-lib.so
+0x0000007d1c4a9000 - 0x0000007d1c4aa000 r-- /data/app/sg.vantagepoint.helloworldjni-1/lib/arm64/libnative-lib.so
+0x0000007d1c4aa000 - 0x0000007d1c4ab000 rw- /data/app/sg.vantagepoint.helloworldjni-1/lib/arm64/libnative-lib.so
+0x0000007d1c516000 - 0x0000007d1c54d000 r-- /data/app/sg.vantagepoint.helloworldjni-1/base.apk
+0x0000007dbd23c000 - 0x0000007dbd247000 r-- /data/app/sg.vantagepoint.helloworldjni-1/base.apk
+0x0000007dc05db000 - 0x0000007dc05dc000 r-- /data/app/sg.vantagepoint.helloworldjni-1/oat/arm64/base.art
+```
+
+While you're searching or exploring the app memory, you can always verify where you're located in each moment (where your current offset is located) in the memory map. Instead of noting and searching for the memory address in this list you can simply run `\dm.`. You'll find an example in the following section "In-Memory Search".
+
+If you're only interested into the modules (binaries and libraries) that the app has loaded, you can use the command `\il` to list them all:
+
+```shell
+[0x00000000]> \il
+0x000000558b1fd000 app_process64
+0x0000007dbc859000 libandroid_runtime.so
+0x0000007dbf5d7000 libbinder.so
+0x0000007dbff4d000 libcutils.so
+0x0000007dbfd13000 libhwbinder.so
+0x0000007dbea00000 liblog.so
+0x0000007dbcf17000 libnativeloader.so
+0x0000007dbf21c000 libutils.so
+0x0000007dbde4b000 libc++.so
+0x0000007dbe09b000 libc.so
+...
+0x0000007d10dd0000 base.odex
+0x0000007d1c499000 libnative-lib.so
+0x0000007d2354e000 frida-agent-64.so
+0x0000007dc065d000 linux-vdso.so.1
+0x0000007dc065f000 linker64
+```
+
+As you might expect you can correlate the addresses of the libraries with the memory maps: e.g. the native library of the app is located at `0x0000007d1c499000` and optimized dex (base.odex) at `0x0000007d10dd0000`.
+
+You can also use objection to display the same information.
+
+```shell
+$ objection --gadget sg.vantagepoint.helloworldjni explore
```

Comment:
maybe make sure you explain somewhere how to get the gadget in... or refer to that chapter where we explain how to use objection earlier?

Reply:
ah it is there, but way earlier. sorry!